src/entity.ts

```diff
@@ -289,6 +289,32 @@ export class Entity {
 	serialize(): object {
 		return this.serializer.serialize(this);
 	}
+
+	persist() {
+		const visited = new Set<Entity>();
+
+		function _persist(entity: Entity) {
+			if (visited.has(entity))
+				return;
+
+			visited.add(entity);
+
+			const identifierProp = entity.meta.type.identifier;
+			if (identifierProp && identifierProp.value(entity) === null)
+				return;
+
+			entity.meta.isNew = false;
+			for (const property of entity.meta.type.properties.filter(p => isEntityType(p.propertyType))) {
+				const value = property.value(entity);
```

Comment:
Should we filter out child entities that have an identifier property in this loop? If the idea is that when an entity is persisted its children that it "owns" are also persisted, then I would think a child with an identifier property (whether it has a value or not) wouldn't be included, as well as its children. Of course, I think the only time this would matter would be if an entity was persisted with a reference to a non-persisted entity. 🤷‍♂️

Reply:
`markPersisted()` no longer traverses nested, identifiable entities.